brainstorming.md

```diff
@@ -82,24 +83,15 @@ SOMACollection is an unordered, `string`-keyed map of values. Values may be any
 
 ### SOMADataFrame
 
-`SOMADataFrame` is a multi-index, multi-column table of monomorphic arrays, all of equal length. A `SOMADataFrame` has a user-defined schema, which defines the number of columns and their respective column name and value type. The schema is expressed as an Arrow `Schema`. All `SOMADataFrame` contain a "pseudo-column" called `__rowid`, of type `uint64` and domain `[0, #rows)`. The `__rowid` pseudo-column contains a unique value for each row in the `SOMADataFrame`, and is intended to act as a join key for other objects, such as a `SOMANdArray`.
-
-A `SOMADataFrame` may be `user-indexed` or `row-indexed`:
-
-- `row-indexed` - each row is addressable (indexed) by the pseudo-column `__rowid` (i.e., a single-index using offset indexing).
-- `user-indexed` - each row is addressable (indexed) by one or more user-specified index columns, which must be named columns in the dataframe schema.
-
-Row-indexed datafames must have no user-specified index columns and at least one column defined in their schema. User-indexed dataframes must have at least one schema-defined column named as an index column. The user-specified schema may not redefine the `__rowid` pseudo-column.
+`SOMADataFrame` is a multi-column table with a user-defined schema, defining the number of columns and their respective column name and value type. The schema is expressed as an Arrow `Schema`. All `SOMADataFrame` contain a "pseudo-column" called `__rowid`, of type `uint64` and domain `[0, #rows)`. The `__rowid` pseudo-column contains a unique value for each row in the `SOMADataFrame`, and is intended to act as a join key for other objects, such as a `SOMADenseNdArray`.
```

Comment:
Just noting here that there are two columns of note:
1. The `__rowid` mentioned here, which is *materialized* and is the "join" column with some *array* dimension values.
2. A virtual "row id" column for the dataframe indicating the positions of the rows in the dataframe, which is *not materialized* and simply facilitates slicing dataframe rows by position.

Not sure if you'd like to mention both here and give different descriptive names to avoid confusion.

Reply:
resolved in the latest spec per our conversation.  Please have a read and see what you think.